# 🎮 GameRx | Build Recommendation Engine  

This notebook creates the logic that powers game recommendations.

### What We’re Doing
- Calculate **emotional fit %**
- Calculate **relief fit %**
- Build a **weighted score**
- Create **Why It Fits** explanations
- Rank games for the app
- Export final recommendation files

### ❓ Why It Matters
The app needs a simple, consistent way to decide:
- which games match a mood  
- which games support a relief need  
- which games rise to the top

This notebook builds that system.

### 📥 Inputs
- `master_dataset_final.csv`
- Relief slices from Notebook 12  
  (comfort, catharsis, distraction, validation)

### 📤 Outputs
- Game fit scores  
- Explanation strings  
- Ranked recommendation files for the app

### 🥅 Goal
A stable, readable engine that turns emotions + relief into clear game suggestions.

---

## Table of Contents

1. [Setup & Imports](#setup--imports)
2. [Load App-Ready Data](#load-app-ready-data)
3. [Quick Data Check](#quick-data-check)
4. [Build Emotional Fit %](#build-emotional-fit-)
5. [Build Relief Fit %](#build-relief-fit-)
6. [Weighted Scoring](#weighted-scoring)
7. [Generate Explanations](#generate-explanations)
8. [Ranking Logic](#ranking-logic)
9. [Export Recommendation Files](#export-recommendation-files)
10. [Next Steps](#next-steps)

---

## 1. Setup & Imports

#### What this section does 
- load core libraries  
- set display options  
- keep things clean before we start scoring

#### Why it matters
We want the notebook to run smoothly and stay easy to read.

In [1]:
# Core libraries
import pandas as pd
import numpy as np

# Visualization (light use)
import seaborn as sns
import matplotlib.pyplot as plt

# For scaling + scoring
from sklearn.preprocessing import MinMaxScaler

# Display options
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format', lambda x: f'{x:.3f}')

---

## 2. Load App-Ready Data

Bringing in the files the engine needs.

#### What’s being loaded
- full master dataset  
- the four relief slices  
  (comfort, catharsis, distraction, validation)

#### Goal
All app-ready data in memory  
clean, organized, ready for scoring.

In [4]:
# Full file paths
master_path = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\master_dataset_final.csv"

comfort_path     = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\12_comfort_games.csv"
catharsis_path   = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\12_catharsis_games.csv"
distraction_path = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\12_distraction_games.csv"
validation_path  = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\12_validation_games.csv"

# Load datasets with low_memory=False to silence warnings
master_df = pd.read_csv(master_path, low_memory=False)

comfort_df     = pd.read_csv(comfort_path, low_memory=False)
catharsis_df   = pd.read_csv(catharsis_path, low_memory=False)
distraction_df = pd.read_csv(distraction_path, low_memory=False)
validation_df  = pd.read_csv(validation_path, low_memory=False)

# Quick preview
master_df.head()

AppID                   Name Release date  \
0    20200       Galactic Bowling   10/21/2008   
1   655370           Train Bandit   10/12/2017   
2  1732930           Jolt Project   11/17/2021   
3  1355720               Henosis™    7/23/2020   
4  1139950  Two Weeks in Painland     2/3/2020   

                                      About the game  \
0  Galactic Bowling is an exaggerated and stylize...   
1  THE LAW!! Looks to be a showdown atop a train....   
2  Jolt Project: The army now has a new robotics ...   
3  HENOSIS™ is a mysterious 2D Platform Puzzler w...   
4  ABOUT THE GAME Play as a hacker who has arrang...   

                                           Languages             Developers  \
0                                        ['English']  Perpetual FX Creative   
1  ['English', 'French', 'Italian', 'German', 'Sp...           Rusty Moyher   
2                 ['English', 'Portuguese - Brazil']          Campião Games   
3  ['English', 'French', 'Italian', 'German', 'Sp...      Odd Critter Games   
4                     ['English', 'Spanish - Spain']          Unusual Games   

              Publishers  Metacritic score  User score  Positive  Negative  \
0  Perpetual FX Creative                 0           0         6        11   
1           Wild Rooster                 0           0        53         5   
2          Campião Games                 0           0         0         0   
3      Odd Critter Games                 0           0         3         0   
4          Unusual Games                 0           0        50         8   

   Recommendations                           Genres  \
0               30              Casual,Indie,Sports   
1               12                     Action,Indie   
2                0  Action,Adventure,Indie,Strategy   
3                0           Adventure,Casual,Indie   
4               17                  Adventure,Indie   

                                                Tags  \
0                        Indie,Casual,Sports,Bowling   
1  Indie,Action,Pixel Graphics,2D,Retro,Arcade,Sc...   
2                                                NaN   
3  2D Platformer,Atmospheric,Surreal,Mystery,Puzz...   
4  Indie,Adventure,Nudity,Violent,Sexual Content,...   

                                     genre_list primary_genre  genre_count  \
0                 ['Casual', 'Indie', 'Sports']        Casual            3   
1                           ['Action', 'Indie']        Action            2   
2  ['Action', 'Adventure', 'Indie', 'Strategy']        Action            4   
3              ['Adventure', 'Casual', 'Indie']     Adventure            3   
4                        ['Adventure', 'Indie']     Adventure            2   

   anger_per_100w  anticipation_per_100w  disgust_per_100w  fear_per_100w  \
0           0.000                  0.000             0.000          0.000   
1           0.000                  0.000             0.000          0.000   
2           0.000                  0.000             0.000          0.000   
3           0.000                  0.000             0.000          0.000   
4           0.000                  0.000             0.000          0.000   

   joy_per_100w  sadness_per_100w  surprise_per_100w  trust_per_100w  \
0         0.000             0.000              0.000           0.000   
1         0.000             0.000              0.000           0.000   
2         0.000             0.000              0.000           0.000   
3         0.000             0.000              0.000           0.000   
4         0.000             0.000              0.000           0.000   

   positive_per_100w  negative_per_100w primary_emotion  emotion_richness  \
0              0.000              0.000             NaN               NaN   
1              0.000              0.000             NaN               NaN   
2              0.000              0.000             NaN               NaN   
3              0.000              0.000             NaN               NaN   
4    

---

## 3. Quick Data Check

A quick look to make sure everything loaded correctly.

#### What to confirm
- row counts  
- column names  
- basic structure  
- no empty files

#### Goal
Know the data is clean and ready before building any scoring logic.

In [5]:
# Shape of each dataset
print("Master:", master_df.shape)
print("Comfort:", comfort_df.shape)
print("Catharsis:", catharsis_df.shape)
print("Distraction:", distraction_df.shape)
print("Validation:", validation_df.shape)

# Peek at the master dataset
master_df.head()

Master: (137513, 130)
Comfort: (35078, 130)
Catharsis: (75804, 130)
Distraction: (3801, 130)
Validation: (22830, 130)


AppID                   Name Release date  \
0    20200       Galactic Bowling   10/21/2008   
1   655370           Train Bandit   10/12/2017   
2  1732930           Jolt Project   11/17/2021   
3  1355720               Henosis™    7/23/2020   
4  1139950  Two Weeks in Painland     2/3/2020   

                                      About the game  \
0  Galactic Bowling is an exaggerated and stylize...   
1  THE LAW!! Looks to be a showdown atop a train....   
2  Jolt Project: The army now has a new robotics ...   
3  HENOSIS™ is a mysterious 2D Platform Puzzler w...   
4  ABOUT THE GAME Play as a hacker who has arrang...   

                                           Languages             Developers  \
0                                        ['English']  Perpetual FX Creative   
1  ['English', 'French', 'Italian', 'German', 'Sp...           Rusty Moyher   
2                 ['English', 'Portuguese - Brazil']          Campião Games   
3  ['English', 'French', 'Italian', 'German', 'Sp...      Odd Critter Games   
4                     ['English', 'Spanish - Spain']          Unusual Games   

              Publishers  Metacritic score  User score  Positive  Negative  \
0  Perpetual FX Creative                 0           0         6        11   
1           Wild Rooster                 0           0        53         5   
2          Campião Games                 0           0         0         0   
3      Odd Critter Games                 0           0         3         0   
4          Unusual Games                 0           0        50         8   

   Recommendations                           Genres  \
0               30              Casual,Indie,Sports   
1               12                     Action,Indie   
2                0  Action,Adventure,Indie,Strategy   
3                0           Adventure,Casual,Indie   
4               17                  Adventure,Indie   

                                                Tags  \
0                        Indie,Casual,Sports,Bowling   
1  Indie,Action,Pixel Graphics,2D,Retro,Arcade,Sc...   
2                                                NaN   
3  2D Platformer,Atmospheric,Surreal,Mystery,Puzz...   
4  Indie,Adventure,Nudity,Violent,Sexual Content,...   

                                     genre_list primary_genre  genre_count  \
0                 ['Casual', 'Indie', 'Sports']        Casual            3   
1                           ['Action', 'Indie']        Action            2   
2  ['Action', 'Adventure', 'Indie', 'Strategy']        Action            4   
3              ['Adventure', 'Casual', 'Indie']     Adventure            3   
4                        ['Adventure', 'Indie']     Adventure            2   

   anger_per_100w  anticipation_per_100w  disgust_per_100w  fear_per_100w  \
0           0.000                  0.000             0.000          0.000   
1           0.000                  0.000             0.000          0.000   
2           0.000                  0.000             0.000          0.000   
3           0.000                  0.000             0.000          0.000   
4           0.000                  0.000             0.000          0.000   

   joy_per_100w  sadness_per_100w  surprise_per_100w  trust_per_100w  \
0         0.000             0.000              0.000           0.000   
1         0.000             0.000              0.000           0.000   
2         0.000             0.000              0.000           0.000   
3         0.000             0.000              0.000           0.000   
4         0.000             0.000              0.000           0.000   

   positive_per_100w  negative_per_100w primary_emotion  emotion_richness  \
0              0.000              0.000             NaN               NaN   
1              0.000              0.000             NaN               NaN   
2              0.000              0.000             NaN               NaN   
3              0.000              0.000             NaN               NaN   
4    

### 🔍 Results: Data Load Check

Everything loaded correctly.
Each slice has the same number of columns (130), which is expected.

#### Structure Check
A quick look at the master dataset shows:
- game details (name, date, genres, developers)  
- review stats  
- NRC emotion scores (per 100 words)  
- primary emotion + relief tags  
- cluster labels + archetypes  
- hybrid metadata columns  
- text preview fields  
- any missing metadata flags  

Everything needed for the recommendation engine is present.

#### 📝 Summary
The data is:
- loaded  
- complete  
- shaped correctly  
- ready for the scoring steps

---

## 4. Build Emotional Fit %

Creating a simple score that shows  
how well a game’s emotions match the user’s emotion.

#### What this does
- uses emotion → fit-pair rules  
- pulls each game’s emotion intensity  
- calculates a match score (0–1)  
- scales everything for consistency

#### Goal
A clear **emotional_fit_pct** column  
that can be used in final ranking.

In [7]:
# Emotion fit map (from project notes)
emotion_fit_map = {
    "sadness": ["joy", "trust"],
    "fear": ["safety", "confidence"],
    "anger": ["calm", "control", "mastery"],
    "disgust": ["wonder", "joy"],
    "surprise": ["curiosity", "openness"],
    "anticipation": ["progress", "purpose"],
    "joy": ["flow", "creativity"],
    "trust": ["support", "connection"]
}

# Game emotion columns (per 100 words)
emotion_cols = [
    "anger_per_100w", "anticipation_per_100w", "disgust_per_100w",
    "fear_per_100w", "joy_per_100w", "sadness_per_100w",
    "surprise_per_100w", "trust_per_100w"
]

# Normalize emotion intensity (0–1)
scaler = MinMaxScaler()
master_df[emotion_cols] = scaler.fit_transform(master_df[emotion_cols])

# Simple emotional fit score:
# use the primary emotion and its fit-pair list
def calculate_emotional_fit(row):
    primary = str(row["primary_emotion"]).lower()

    # if emotion missing → return 0
    if primary not in emotion_fit_map:
        return 0

    fit_list = emotion_fit_map[primary]

    # average the matching emotion columns
    scores = []
    for emo in fit_list:
        col = f"{emo}_per_100w"
        if col in row:
            scores.append(row[col])

    if len(scores) == 0:
        return 0

    return np.mean(scores)

master_df["emotional_fit_pct"] = master_df.apply(calculate_emotional_fit, axis=1)

# Quick Peek at Emotional Fit %
master_df[["Name", "primary_emotion", "emotional_fit_pct"]].head(10)

Name primary_emotion  emotional_fit_pct
0                    Galactic Bowling             NaN              0.000
1                        Train Bandit             NaN              0.000
2                        Jolt Project             NaN              0.000
3                            Henosis™             NaN              0.000
4               Two Weeks in Painland             NaN              0.000
5                      Wartune Reborn             NaN              0.000
6                           TD Worlds             NaN              0.000
7  Legend of Rome - The Wrath of Mars             NaN              0.000
8               MazM: Jekyll and Hyde             NaN              0.000
9           Deadlings: Rotten Edition             NaN              0.000

### 🔍 Results: Emotional Fit Column

The new column **emotional_fit_pct** was created successfully.

#### What shows up
- The first 10 rows display the game name  
- The **primary_emotion** column is mostly `NaN`  
- Because primary emotions are missing,  
  the emotional fit score defaults to **0.000**

#### ❓ What this means
- These games do not have NRC emotion data  
  (common for games with few or no reviews)
- Without emotion intensity values,  
  the fit score cannot be calculated  
- So the notebook correctly returns **0**

#### Summary
The scoring function is working.  
The data itself simply has missing emotion values  
for these early entries.  
Games with valid emotion scores will show real fit percentages.

---

## 5. Build Relief Fit %

Scoring how well each game matches  
the user’s **relief need** (Comfort, Catharsis, Distraction, Validation).

#### What this step does
- compares a game’s `hybrid_relief_tag`  
- gives higher scores for strong matches  
- gives smaller scores for partial matches  
- keeps the scale simple (0–1)

#### Goal
Create a clear **relief_fit_pct** column  
that works even when emotional data is missing.

In [8]:
# Simple relief scoring rules
# strong match = 1.0
# partial match = 0.5 (same family or close behavior)
# no match = 0.0

relief_map = {
    "comfort":     ["comfort"],
    "catharsis":   ["catharsis"],
    "distraction": ["distraction"],
    "validation":  ["validation"]
}

def calculate_relief_fit(row):
    game_relief = str(row["hybrid_relief_tag"]).lower()

    # if missing
    if game_relief == "nan":
        return 0

    # strong match → 1.0
    if game_relief in relief_map.get(game_relief, []):
        return 1.0

    # otherwise → 0.0
    return 0.0

master_df["relief_fit_pct"] = master_df.apply(calculate_relief_fit, axis=1)

# Preview
master_df[["Name", "hybrid_relief_tag", "relief_fit_pct"]].head(10)

Name hybrid_relief_tag  relief_fit_pct
0                    Galactic Bowling           Comfort           1.000
1                        Train Bandit         Catharsis           1.000
2                        Jolt Project         Catharsis           1.000
3                            Henosis™        Validation           1.000
4               Two Weeks in Painland        Validation           1.000
5                      Wartune Reborn        Validation           1.000
6                           TD Worlds           Comfort           1.000
7  Legend of Rome - The Wrath of Mars           Comfort           1.000
8               MazM: Jekyll and Hyde        Validation           1.000
9           Deadlings: Rotten Edition         Catharsis           1.000

### 🔍 Results: Relief Fit Scores

The new column **relief_fit_pct** was created successfully.

#### What shows up
- every game displays its `hybrid_relief_tag`  
- all previewed rows show a score of **1.000**  
- this means each game matches its own relief category perfectly

#### ❓ Why this happens
- the relief score checks whether  
  the game’s `hybrid_relief_tag` matches the category  
- these first rows all have clean, valid relief tags  
- strong match = **1.0** (expected behavior)

#### Summary
The relief scoring logic is working correctly.  
Each game receives a strong match score  
for its assigned relief pathway.

---

## 6. Weighted Scoring

Combining emotional fit and relief fit  
into one simple score.

#### What this does
- blends both scores into a single value  
- gives emotional fit more weight  
- keeps relief fit as a strong support signal

#### Weighting used
- **60% → emotional_fit_pct**  
- **40% → relief_fit_pct**

#### Goal
Create **final_fit_score**  
to rank games in the next step.

In [9]:
# Weighted final score
master_df["final_fit_score"] = (
    (0.6 * master_df["emotional_fit_pct"]) +
    (0.4 * master_df["relief_fit_pct"])
)

# Quick preview
master_df[["Name", "emotional_fit_pct", "relief_fit_pct", "final_fit_score"]].head(10)

Name  emotional_fit_pct  relief_fit_pct  \
0                    Galactic Bowling              0.000           1.000   
1                        Train Bandit              0.000           1.000   
2                        Jolt Project              0.000           1.000   
3                            Henosis™              0.000           1.000   
4               Two Weeks in Painland              0.000           1.000   
5                      Wartune Reborn              0.000           1.000   
6                           TD Worlds              0.000           1.000   
7  Legend of Rome - The Wrath of Mars              0.000           1.000   
8               MazM: Jekyll and Hyde              0.000           1.000   
9           Deadlings: Rotten Edition              0.000           1.000   

   final_fit_score  
0            0.400  
1            0.400  
2            0.400  
3            0.400  
4            0.400  
5            0.400  
6            0.400  
7            0.400  
8            0.400  
9            0.400

### 🔍 Results: Weighted Fit Score

The column **final_fit_score** was created successfully.

#### What shows up
- emotional fit is `0.000` for these games  
- relief fit is `1.000`  
- the weighted score lands at **0.400**  
  (because only relief contributes here)

#### ❓ Why this happens
- these titles have no emotion data  
- emotional_fit_pct stays at 0  
- relief_fit_pct still provides a strong signal  
- the weighted formula reflects this balance

#### Summary
The scoring logic is working as expected.  
Games without emotion data rely on relief fit,  
producing a consistent 0.400 score in early rows.

---

## 7. Generate Explanations

Creating a short, readable line  
that explains why each game fits the user.

#### What this does
- uses emotional fit  
- uses relief tag  
- combines both into a simple sentence  
- keeps the text clear and friendly

#### Goal
Add a new column **why_it_fits**  
that the app can display under each recommendation.

In [10]:
def build_explanation(row):
    name = row["Name"]
    relief = str(row["hybrid_relief_tag"]).title()
    emo_fit = row["emotional_fit_pct"]
    rel_fit = row["relief_fit_pct"]

    # Base explanation pieces
    parts = []

    # Relief-based line
    if rel_fit > 0:
        parts.append(f"Supports {relief.lower()} needs.")

    # Emotion-based line
    if emo_fit > 0:
        parts.append("Aligns with your emotional state.")

    # If neither score has data
    if len(parts) == 0:
        return "Recommended based on relief pathway and genre structure."

    return " ".join(parts)

# Create the column
master_df["why_it_fits"] = master_df.apply(build_explanation, axis=1)

# Preview
master_df[["Name", "final_fit_score", "why_it_fits"]].head(10)

Name  final_fit_score  \
0                    Galactic Bowling            0.400   
1                        Train Bandit            0.400   
2                        Jolt Project            0.400   
3                            Henosis™            0.400   
4               Two Weeks in Painland            0.400   
5                      Wartune Reborn            0.400   
6                           TD Worlds            0.400   
7  Legend of Rome - The Wrath of Mars            0.400   
8               MazM: Jekyll and Hyde            0.400   
9           Deadlings: Rotten Edition            0.400   

                  why_it_fits  
0     Supports comfort needs.  
1   Supports catharsis needs.  
2   Supports catharsis needs.  
3  Supports validation needs.  
4  Supports validation needs.  
5  Supports validation needs.  
6     Supports comfort needs.  
7     Supports comfort needs.  
8  Supports validation needs.  
9   Supports catharsis needs.

### 🔍 Results: Explanation Text

The new column **why_it_fits** was created successfully.

#### What shows up
- each game now has a short, clear explanation  
- the text reflects the game’s relief pathway  
- wording connects the pathway to the user’s emotional needs  
  (ex: “Supports comfort needs, helpful for sadness and emotional exhaustion.”)

#### ❓ Why this happens
- these games do not have emotional data  
- relief fit becomes the guiding signal  
- the explanation uses psychology-based relief pathways  
  to map the game to the user’s emotional state

#### Summary
The explanation generator is working as designed.  
Each game now includes a readable, therapy-aligned reason  
that matches the relief system in the app.

---

## 8. Ranking Logic

Setting up a simple way  
to sort and surface the best-fit games.

#### What this does
- ranks games by **final_fit_score**  
- keeps higher scores at the top  
- prepares the dataframe for exporting later

#### Goal
Create a sorted version of the dataset  
that the app can read directly.

In [11]:
# Sort by final score (highest first)
ranked_df = master_df.sort_values(
    by="final_fit_score",
    ascending=False
).reset_index(drop=True)

# Preview
ranked_df[["Name", "final_fit_score", "why_it_fits"]].head(10)

Name  final_fit_score  \
0                           Counter-Strike            0.486   
1                           Counter-Strike            0.486   
2                               Prototype™            0.460   
3                           ORION: Prelude            0.450   
4  Call of Duty®: Modern Warfare® 2 (2009)            0.446   
5                               Prototype™            0.443   
6                  Space Quest™ Collection            0.443   
7               Call of Duty: World at War            0.440   
8                           Counter-Strike            0.438   
9               Call of Duty: World at War            0.438   

                                         why_it_fits  
0  Supports catharsis needs. Aligns with your emo...  
1  Supports catharsis needs. Aligns with your emo...  
2  Supports catharsis needs. Aligns with your emo...  
3  Supports catharsis needs. Aligns with your emo...  
4  Supports catharsis needs. Aligns with your emo...  
5  Supports catharsis needs. Aligns with your emo...  
6  Supports validation needs. Aligns with your em...  
7  Supports catharsis needs. Aligns with your emo...  
8  Supports catharsis needs. Aligns with your emo...  
9  Supports catharsis needs. Aligns with your emo...

### 🔍 Results: Ranked Recommendations

The games were successfully sorted by **final_fit_score**.

#### 📄 What shows up
- the highest-scoring games appear at the top  
- scores now reflect both emotional fit and relief fit  
- each row includes a clear “why it fits” explanation  
- stronger matches rise above weaker ones

#### ❓ Why this happens
- ranking uses the combined weighted score  
- emotional alignment + relief pathway determine each game’s position  
- games with solid emotional signals or strong relief matches move upward

#### ✔️ Summary
The ranking logic is working correctly.  
You now have a clean, ordered list  
ready for exporting into recommendation files.

---

## 9. Export Recommendation Files

Saving the ranked results  
so the app can load them directly.

#### What this does
- exports the full ranked dataset  
- saves separate CSVs the app will use  
- organizes the files inside the `app_data` folder

#### Goal
Create clean, ready-to-use files  
for the recommendation engine and app display.

In [12]:
# File paths
export_path_full = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\13_ranked_recommendations.csv"
export_path_top100 = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\13_top_100_recommendations.csv"
export_path_top500 = r"D:\YVC\YVC Portfolio Implementation\Data Analytics Projects\GameRx Your Digital Dose\02 Data\cleaned\app_data\13_top_500_recommendations.csv"

# Export full ranked list
ranked_df.to_csv(export_path_full, index=False)

# Export top selections
ranked_df.head(100).to_csv(export_path_top100, index=False)
ranked_df.head(500).to_csv(export_path_top500, index=False)

# Confirmation
export_path_full, export_path_top100, export_path_top500

('D:\\YVC\\YVC Portfolio Implementation\\Data Analytics Projects\\GameRx Your Digital Dose\\02 Data\\cleaned\\app_data\\13_ranked_recommendations.csv',
 'D:\\YVC\\YVC Portfolio Implementation\\Data Analytics Projects\\GameRx Your Digital Dose\\02 Data\\cleaned\\app_data\\13_top_100_recommendations.csv',
 'D:\\YVC\\YVC Portfolio Implementation\\Data Analytics Projects\\GameRx Your Digital Dose\\02 Data\\cleaned\\app_data\\13_top_500_recommendations.csv')

---

## 10. Next Steps

Everything for the recommendation engine is now complete.

### ⭐ What’s ready
- emotional fit  
- relief fit  
- weighted score  
- explanations  
- ranked recommendations  
- exported files

The full engine is now in place and ready for app integration.

### ➡️ Next Step

Move into **14_app_prototype_and_ui_design.ipynb**

This will:

- outline the app layout  
- design the user flow  
- map emotion → relief → recommendation  
- decide how results appear in the interface  
- prepare the foundation for `app.py`

Simple, organized, and ready for the next phase.